warehouse/query-core/src/test/java/datawave/query/transformer/SSDeepSimilarityQueryTransformerTest.java

```diff
@@ -0,0 +1,95 @@
+package datawave.query.transformer;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.api.easymock.annotation.Mock;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import com.google.common.collect.Multimap;
+import com.google.common.collect.TreeMultimap;
+
+import datawave.marking.MarkingFunctions;
+import datawave.query.tables.SSDeepSimilarityQueryLogic;
+import datawave.query.util.ssdeep.NGramTuple;
+import datawave.query.util.ssdeep.SSDeepHash;
+import datawave.webservice.query.Query;
+import datawave.webservice.query.result.event.DefaultEvent;
+import datawave.webservice.query.result.event.DefaultField;
+import datawave.webservice.query.result.event.ResponseObjectFactory;
+import datawave.webservice.result.BaseQueryResponse;
+import datawave.webservice.result.DefaultEventQueryResponse;
+
+@RunWith(PowerMockRunner.class)
+public class SSDeepSimilarityQueryTransformerTest {
+    private SSDeepSimilarityQueryTransformer transformer;
+
+    @Mock
+    private Query mockQuery;
+
+    @Mock
+    private MarkingFunctions mockMarkingFunctions;
+
+    @Mock
+    private ResponseObjectFactory mockResponseFactory;
+
+    String chunk = "//thPkK";
+    int chunkSize = 3;
+    String ssdeepString = "3:yionv//thPkKlDtn/rXScG2/uDlhl2UE9FQEul/lldDpZflsup:6v/lhPkKlDtt/6TIPFQEqRDpZ+up";
+
+    public void basicExpects(Key k) {
+        NGramTuple tuple = new NGramTuple(chunkSize, chunk);
+        SSDeepHash hash = SSDeepHash.parse(ssdeepString);
+        Multimap<NGramTuple,SSDeepHash> queryMap = TreeMultimap.create();
+        queryMap.put(tuple, hash);
+        List<String> encodedQueryMap = SSDeepSimilarityQueryLogic.encodeQueryMap(queryMap);
+        Map<String,List<String>> optionalParameters = new HashMap<>();
+        optionalParameters.put(SSDeepSimilarityQueryLogic.SSDEEP_QUERY_MAP_PARAM, encodedQueryMap);
+
+        EasyMock.expect(mockQuery.getQueryAuthorizations()).andReturn("A,B,C");
+        EasyMock.expect(mockQuery.getOptionalQueryParameters()).andReturn(optionalParameters);
+        EasyMock.expect(mockResponseFactory.getEventQueryResponse()).andReturn(new DefaultEventQueryResponse());
+        EasyMock.expect(mockResponseFactory.getEvent()).andReturn(new DefaultEvent()).times(1);
+        EasyMock.expect(mockResponseFactory.getField()).andReturn(new DefaultField()).times(4);
+    }
+
+    @Test
+    public void transformTest() {
```

Comment:
I was unable to get the transform unit tests to run. I kept getting powermock related errors related to javax.management. This is most likely a local environment issue on my end.

Reply:
Frankly, I'm surprised it worked in CI/CD. Nevertheless, I added the following line just about the class declaration, and this causes powermock to ignore `javax.management`. `javax.xml` also must be ignored. I wonder if newer versions of powermock do this by default.

```
@PowerMockIgnore({"javax.management.*", "javax.xml.*"})
```